README.md

```diff
@@ -17,3 +15,21 @@ The vizlab wiki at https://github.com/USGS-VIZLAB/vizlab/wiki has help pages on
 * Collaborating
 * Using vizlab
 * Specific concepts (timestamps and SVG inject)
+
+# Docker
+ This project now has a `Dockerfile` included, to try out Docker to standardize our development and build environments.  There are a few simple steps to build the docker **image** , start up the docker **container**, and then set up the repository and Rstudio project as you normally would.  
+ 
+## Docker basics
+  The Dockerfile is a text file that contains the instructions to build the docker **image** --- a layered binary file that contains everything you installed. The image has everything needed for docker to start up a docker **container**, inside of which you run the commands/programs you want to use. Once a container is created and has run what you need it to, you can stop the container, and either remove it or restart it later.  Containers are meant to be ephemeral --- anything you create in the container that is important should either be scripted so it can be recreated, or saved to a **volume**.  Volumes are docker storage centers on your hard drive that persist beyond the life of a container, but they are _not_ normal directories that you can access outside of docker.  
+  
+  Docker commands all either start with `docker` or `docker-compose`.  They can generally do the same things, but `docker` accepts command-line flags (e.g. `docker run -t tag`) while `docker-compose` uses the `docker-compose.yml` for options.  We mostly use `docker-compose` here so the options can be easily source-controlled.
+
+## To build/work on a viz
+  First, either clone your fork of the repo, or pull down only the `Dockerfile` and `docker-compose.yml`.  In a terminal, go into the directory containing the `Dockerfile` and `docker-compose.yml`, and run `docker-compose build`.  This builds the docker image using the options in `docker-compose.yml`.  Next, run `docker-compose up`.  This creates and starts the docker container and leaves it running, with Rstudio exposed on port 8787.  Go to your web browser and you can log in to Rstudio at `localhost:8787`.  The username is `rstudio`, and password is set in the `docker-compose.yml`.  Now you can use Rstudio as you would normally, i.e. creating a new project from a Github repo.  Note the container does not contain any of your credentials, for Github or elsewhere.  However, the container already contains the DOI root certificate, so HTTPS will work over the network. (Note that you should **not** upload an **image** containing the DOI cert to a public repo.) Files you save in Rstudio will be contained in a docker volume, and will persist beyond the life of the container (unless you delete the volume, of course).  When you are done, log out of Rstudio and run `docker-compose down` in your terminal to stop the docker container.      
```

Comment:
Agree, although I think we should probably just clone the repo to get the `Dockfile` and `docker-compose.yml` since it is fewer clicks/commands, and they might change during the course of a project

Reply:
Fewer clicks/commands is a worthy consideration. However, I'm worried about the temptation to edit files in this local directory rather than editing them within the docker container, especially as our team is just getting started with this tool. If we don't keep a full copy of the repo locally, there won't be much temptation. So instead of git cloning the entire repo, how about a sparse checkout a la https://stackoverflow.com/questions/600079/how-do-i-clone-a-subdirectory-only-of-a-git-repository/13738951#13738951? I think it would look like this:
```sh
mkdir water-use-15-docker
cd water-use-15-docker
git init
git remote add -f origin git@github.com:wdwatkins/water-use-15.git
git config core.sparseCheckout true
echo Dockerfile >> .git/info/sparse-checkout
echo docker-compose.yml >> .git/info/sparse-checkout
git pull origin master
```
and from there, even though there are only those two files in your local folder, you can still git commit, push, and pull as usual (for those two files).

Reply:
Mm interesting.  We could just make that into a bash script then, with an argument for directory name and repo name maybe, so we don't have to run all those commands by hand.

Reply:
It'd be nice to avoid the need for everyone to type this, agreed. But I'm not sure there's much point to creating it as a file within this repo, exactly, since then the user would need to decide how to download this one script (clone, sparse-checkout, or manual download?), which is basically the question we're trying to preempt for everybody by suggesting these bash commands. Would the above snippet within the README, or maybe a gist, suffice?

Reply:
Right, not in this repo, but somewhere.  Link to a gist in the readme maybe?

Reply:
Sure, that seems fine.